src/Compilers/CSharp/Test/Emit/PDB/PDBLambdaTests.cs

```diff
@@ -1381,5 +1381,283 @@ .maxstack  1
 }
 ", sequencePoints: "C+<>c.<F>b__0_0");
         }
+
+        [Fact]
+        public void WithExpression()
+        {
+            var source = MarkedSource(WithWindowsLineBreaks(@"
+using System;
+record R(int X);
+
+class Test
+{
+    public static void M(int a)
+    <M><C:0>{
+        var x = new R(1);
+        var y = x with
+        {
+            X = new Func<int>(() => <L:0.0>a)()
+        };
+    }
+}
+"), removeTags: true); // We're validating offsets so need to remove tags entirely
+
+            // Use NetCoreApp in order to use records
+            var compilation = CreateCompilation(source.Tree, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugDll);
+
+            compilation.VerifyPdbLambdasAndClosures(source);
+        }
+
+        [Fact]
+        public void WithExpression_2()
+        {
+            var source = MarkedSource(WithWindowsLineBreaks(@"
+using System;
+record R(int X, int Y);
+
+class Test
+{
+    public static void M(int a)
+    <M><C:0>{
+        var x = new R(1, 2);
+        var b = 1;
+        var y = x with
+        {
+            X = new Func<int>(() => <L:0.0>a)(),
+            Y = new Func<int>(() => <L:1.0>b)()
+        };
+    }
+}
+"), removeTags: true); // We're validating offsets so need to remove tags entirely
+
+            // Use NetCoreApp in order to use records
+            var compilation = CreateCompilation(source.Tree, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugDll);
+
+            compilation.VerifyPdbLambdasAndClosures(source);
+        }
+
+        [Fact]
+        public void WithExpression_3()
```

Comment:
This test fails IOperation validation in CI on one leg: https://dev.azure.com/dnceng/public/_build/results?buildId=1046502&view=ms.vss-test-web.build-test-results-tab&runId=32377276&resultId=129370&paneView=debug

@jcouv @333fred do you want a bug filed? Or have I done something wrong?

Reply:
Please file a bug and skip this test with `ConditionalFact(typeof(NoIOperationValidation), Reason = "link")`